### PR TITLE
Fix Python plugin

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Fix VS code Python plugin, by correctly building its extension

### CODE REVIEW

File: dependencies/che-plugin-registry/Jenkinsfile-single [hunk 1]
```diff
@@ -73,6 +73,16 @@ def buildPhpDebug(branchToBuildPlugin, extensionFolder) {
     sh "cd ${extensionFolder} && sudo npm install -g vsce gulp && npm install && npm run build && vsce package"
 }
 
+def buildVscodePython(branchToBuildPlugin, extensionFolder) {
+    installNPM()
+    sh "sudo /usr/bin/python3 -m pip install --upgrade pip"
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/Jenkinsfile-single]: probably don't need this any more as it's implemented in the config for the rhel7 nodes:

https://gitlab.cee.redhat.com/codeready-workspaces/cloud-slaves/blob/master/label/linux/pip.sh

But doing it again here won't hurt anything and makes it clear we need latest pip.

File: dependencies/che-plugin-registry/Jenkinsfile-single [hunk 2]
```diff
@@ -73,6 +73,16 @@ def buildPhpDebug(branchToBuildPlugin, extensionFolder) {
...
+    buildNumber = branchToBuildPlugin.substring(branchToBuildPlugin.lastIndexOf('.') + 1)
+    echo "${buildNumber}"
+    sh "cd ${extensionFolder} && sudo npm install -g vsce gulp && npm ci && gulp updateBuildNumber --buildNumber ${buildNumber} && rm -rf node_modules"
+    archiveSources(extensionFolder)
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/Jenkinsfile-single]: you're archiving after building? 

for other extensions, you're archiving before the build. Why do it afterwards here?

File: dependencies/che-plugin-registry/Jenkinsfile-single [hunk 3]
```diff
@@ -148,8 +158,10 @@ timeout(120) {
             buildEclipseCdt(extensionFolder)
         } else if (extensionPath.contains("https://github.com/llvm/llvm-project")) {
             buildClangVscode(extensionFolder)
+        } else if (extensionPath.contains("https://github.com/microsoft/vscode-python")) {
+            buildVscodePython(branchToBuildPlugin, extensionFolder)
         } else if (extensionPath.contains("https://github.com/microsoft/vscode-java-debug")) {
-            buildJavaExtension(extensionFolder, extensionFolder)
+            buildJavaExtension(branchToBuildPlugin, extensionFolder)
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/Jenkinsfile-single]: oops! good typo fix :)

File: dependencies/che-plugin-registry/Jenkinsfile-single (hunk 2 above)
Comment (mkuznyetsov) [reply dependencies/che-plugin-registry/Jenkinsfile-single]: that's not the build, but required things to update the build number (18875) into plugin's package.json (and resulting artifact name). Then in this command I would node_modules (to save some hundreds of MBs) and package my sources. Only after that I would "npm ci" again and perform the build.

**Pull request merged**